onnxruntime/core/providers/nnapi/nnapi_builtin/builders/op_builder_helpers.cc

```diff
@@ -47,7 +61,42 @@
                        const std::string& output, const Shape* output_shape) {
   if (output_shape == nullptr) {
     auto& shaper = model_builder.GetShaper();
-    ORT_RETURN_IF_ERROR(shaper.Reshape(data_input, shape_value, output));
+    auto calculate_reshape_output_shape = [&](const std::string& data_input,
+                                              const std::vector<int32_t>& shape_value,
+                                              const std::string& output) -> Status {
+      const Shape& input_dimen = shaper[data_input];
+      uint32_t input_size = Product(input_dimen);
+      std::vector<uint32_t> output_dimen(shape_value.size());
```

Comment:
>std::vector<uint32_t> outpu [](http://example.com/codeflow?start=6&length=27)

InlinedVector or TensorShapeVector per conding standards.
Do no resize the vector only to throw away the values. Use reserve() + push_back()